src/NHibernate/Hql/Ast/ANTLR/HqlSqlWalker.cs

```diff
@@ -749,7 +759,10 @@ IASTNode CreateFromFilterElement(IASTNode filterEntity, IASTNode alias)
 			{
 				log.Debug("createFromFilterElement() : processed filter FROM element.");
 			}
-	
+
+			if (fromElementFound)
+				return (IASTNode) adaptor.Nil();
```

Comment:
Has this been done because some of the calls above would have side effects? If yes, maybe a comment would be welcome. Otherwise I would tend to move this `return` in the `if (fromElement == null)` clause.